react/features/overlay/components/PageReloadOverlay.js

```diff
@@ -0,0 +1,298 @@
+/* global APP, AJS */
+
+import React, { Component } from 'react';
+
+import { randomInt } from '../../base/util';
+
+import AbstractOverlay from './AbstractOverlay';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * Implements a React Component for the reload timer. Starts counter from
+ * props.start, adds props.step to the current value on every props.interval
+ * seconds until the current value reaches props.end. Also displays progress
+ * bar.
+ */
+class ReloadTimer extends Component {
+    /**
+     * ReloadTimer component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The end of the timer. When this.state.current reaches this
+         * value the timer will stop and call onFinish function.
+         * @public
+         * @type {number}
+         */
+        end: React.PropTypes.number,
+
+        /**
+         * The interval in sec for adding this.state.step to this.state.current
+         * @public
+         * @type {number}
+         */
+        interval: React.PropTypes.number,
+
+        /**
+         * The function that will be executed when timer finish (when
+         * this.state.current === this.props.end)
+         */
+        onFinish: React.PropTypes.func,
+
+        /**
+         * The start of the timer. The initial value for this.state.current.
+         * @public
+         * @type {number}
+         */
+        start: React.PropTypes.number,
+
+        /**
+         * The value which will be added to this.state.current on every step.
+         * @public
+         * @type {number}
+         */
+        step: React.PropTypes.number
+    }
+
+    /**
+     * Initializes a new ReloadTimer instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     * @public
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            current: this.props.start,
+            time: Math.abs(this.props.end - this.props.start)
+        };
+    }
+
+    /**
+     * React Component method that executes once component is mounted.
+     *
+     * @inheritdoc
+     * @returns {void}
+     * @protected
+     */
+    componentDidMount() {
+        AJS.progressBars.update('#reloadProgressBar', 0);
+        const intervalId = setInterval(() => {
```

Comment:
You're starting a timer when the Component is mounted and letting it continue even after the Component is unmounted. While in the particular use of `ReloadTimer` that may not be a problem, it's safer and cleaner in general to clear the timer when the Component unmounts.